modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/user/impl/FileBasedUserStoreConnector.java

```diff
@@ -88,15 +89,15 @@ public UserStoreManagerImpl() throws ConfigurationException {
      * @throws BrokerAuthException Exception throws when authentication failed.
      */
     @Override
-    public boolean authenticate(String userName, char... credentials) throws BrokerAuthException {
+    public AuthResult authenticate(String userName, char... credentials) throws BrokerAuthException {
         User user;
         if (Objects.isNull(userName)) {
             throw new BrokerAuthException("Username cannot be null.");
         } else if (Objects.isNull(user = userRegistry.get(userName))) {
             throw new BrokerAuthException("User not found for the given username.");
         } else {
             if (credentials != null && Arrays.equals(credentials, user.getPassword())) {
```

Comment:
Do we need a null check here? Doesn't Arrays.equals handle this?

Reply:
Fixed in #328